storage/tests/SMAIndexSubBlock_unittest.cpp

```diff
@@ -0,0 +1,998 @@
+    ScalarAttribute *scalar_attribute = new ScalarAttribute(*relation_->getAttributeById(attribute));
+    ScalarLiteral *scalar_literal
+        = new ScalarLiteral(AttributeType::InstanceNonNullable().makeValue(&literal),
+                            AttributeType::InstanceNonNullable());
+    return new ComparisonPredicate(ComparisonFactory::GetComparison(comp), scalar_attribute, scalar_literal);
+  }
+
+  ComparisonPredicate* generateStringComparisonPredicate(
+      const ComparisonID comp,
+      const attribute_id attribute,
+      const string &literal) {
+    ScalarAttribute *scalar_attribute = new ScalarAttribute(*relation_->getAttributeById(attribute));
+    ScalarLiteral *scalar_literal = new ScalarLiteral(
+        VarCharType::InstanceNonNullable(literal.size()).makeValue(
+            literal.c_str(),
+            literal.size() + 1).ensureNotReference(),
+        VarCharType::InstanceNonNullable(literal.size()));
+    return new ComparisonPredicate(ComparisonFactory::GetComparison(comp), scalar_attribute, scalar_literal);
+  }
+
+  // Retrieves the SMAEntry from the test SMAIndex instance.
+  SMAEntry* getEntryForAttribute(attribute_id attribute) {
+    return const_cast<SMAEntry*>(index_->getEntryChecked(attribute));
+  }
+
+  std::unique_ptr<CatalogRelation> relation_;
+  std::unique_ptr<MockTupleStorageSubBlock> tuple_store_;
+  ScopedBuffer index_memory_;
+  std::unique_ptr<IndexSubBlockDescription> index_description_;
+  std::unique_ptr<SMAIndexSubBlock> index_;
+};
+
+const char SMAIndexSubBlockTest::kCharAttrNullValue[] = "_NULLSTRING";
+
+TEST_F(SMAIndexSubBlockTest, DescriptionIsValidTest) {
+  std::unique_ptr<IndexSubBlockDescription> index_description_;
+  vector<attribute_id> valid_attrs({0, 1, 2, 3, 4, 5, 6});
+  // Try to create an index on each of the attributes. Make sure that each
+  // of the attributes which can be indexed on is marked valid.
+  for (const CatalogAttribute &attr : *relation_) {
+    index_description_.reset(
+        new IndexSubBlockDescription());
+    index_description_->set_sub_block_type(
+        IndexSubBlockDescription::SMA);
+    index_description_->AddExtension(
+        SMAIndexSubBlockDescription::indexed_attribute_id,
+        attr.getID());
+    if (std::find(valid_attrs.begin(), valid_attrs.end(), attr.getID()) != valid_attrs.end()) {
+      EXPECT_TRUE(SMAIndexSubBlock::DescriptionIsValid(
+          *relation_,
+          *index_description_))
+               << "Expected attribute " << attr.getID() << " to be valid.";
+    } else {
+      EXPECT_FALSE(SMAIndexSubBlock::DescriptionIsValid(
+          *relation_,
+          *index_description_))
+              << "Expected attribute " << attr.getID() << " to be invalid.";
+    }
+  }
+}
+
+TEST_F(SMAIndexSubBlockTest, TestConstructor) {
+  // Use the test fixture class to create a proper description.
+  createIndex({0, 1, 2}, kIndexSubBlockSize);
+  memset(index_memory_.get(), 0, kIndexSubBlockSize);
```

Comment:
This is a redundant hold over from a previous version of this test. Good catch. Going to remove this line.